pkg/providers/tinkerbell/hardware/validator.go

```diff
@@ -289,8 +275,13 @@ func MatchingDisksForSelectors(selectors []MachineSelector) MachineAssertion {
 	}
 }
 
-// labelsMatchSelector ensures all selector key-value pairs can be found in labels.
-func labelsMatchSelector(selector MachineSelector, labels Labels) bool {
+// LabelsMatchSelector ensures all selector key-value pairs can be found in labels.
+// If selector is empty true is always returned.
+func LabelsMatchSelector(selector v1alpha1.HardwareSelector, labels Labels) bool {
+	if selector.IsEmpty() {
+		return true
+	}
```

Comment:
does this mean we are supporting a case where the labels are not specified and we pick hardware at random?
Based on [the conversation](https://github.com/aws/eks-anywhere/pull/2300#discussion_r889534093) on the label selector PR, it seems like the labels are required

Reply:
This is removed, albeit it was never required. The behavior is the same without it 🤦🏻 .